pkg/generate/ack/controller.go

```diff
@@ -102,7 +103,18 @@ var (
 			return code.SetSDKForStruct(r.Config(), r, targetFieldName, targetVarName, targetShapeRef, sourceFieldPath, sourceVarName, indentLevel)
 		},
 		"GoCodeSetResourceForStruct": func(r *ackmodel.CRD, targetFieldName string, targetVarName string, targetShapeRef *awssdkmodel.ShapeRef, sourceVarName string, sourceShapeRef *awssdkmodel.ShapeRef, indentLevel int) string {
-			return code.SetResourceForStruct(r.Config(), r, targetFieldName, targetVarName, targetShapeRef, sourceVarName, sourceShapeRef, indentLevel)
+			f, ok := r.Fields[targetFieldName]
+			if !ok {
+				return ""
+			}
+			// We may have some special instructions for how to handle setting the
+			// field value...
+			setCfg := f.GetSetterConfig(model.OpTypeList)
```

Comment:
Why `model.OpTypeList` as operation? 
I think adding a comment for the same will be useful as well.